annotation/tests/test_annotations.py

```diff
@@ -235,3 +278,253 @@ def test_create_manifest_json(
+
+
+def test_construct_annotated_doc_db_error(
+    annotated_doc: AnnotatedDoc,
+    annotation_doc_for_save: DocForSaveSchema,
+):
+    is_latest = False
+    new_equal_latest = False
+    doc = annotation_doc_for_save
+    latest_doc = annotated_doc
+    s3_file_path = "path"
+    s3_file_bucket = "bucket"
+
+    pages_sha_latest = ({"1": "a"}, "b")
+    pages_sha_non_latest = ({"1": "a"}, "c")
+
+    db = Mock()
+    db.add = Mock()
+    db.add_all = Mock()
+    db.commit = Mock(
+        side_effect=IntegrityError(
+            statement="TEST", params=("testuser",), orig=Exception("TESTING")
+        )
+    )
+    db.rollback = Mock()
+    with patch(
+        "annotation.annotations.main.get_pages_sha",
+        return_value=pages_sha_latest if is_latest else pages_sha_non_latest,
+    ), patch(
+        "annotation.annotations.main.check_docs_identity",
+        return_value=new_equal_latest,
+    ), patch(
+        "annotation.annotations.main.construct_document_links", return_value=[]
+    ):
+        with pytest.raises(DuplicateAnnotationError):
+            construct_annotated_doc(
+                db,
+                annotated_doc.user,
+                None,
+                annotated_doc.job_id,
+                annotated_doc.file_id,
+                doc,
+                TEST_TENANT,
+                s3_file_path,
+                s3_file_bucket,
+                latest_doc,
+                1,
+                is_latest,
+            )
+            assert db.commit.assert_called_once()
+            assert db.rollback.assert_called_once()
+
+
+def test_upload_json_to_minio():
+    test_json = json.dumps({"test": 1})
+    path_to_object = "path"
+    bucket_name = TEST_TENANT
+
+    with patch(
+        "annotation.annotations.main.bd_storage.get_storage"
+    ) as mock_get_storage:
+        mock_get_storage.return_value.upload_obj = Mock()
+
+        upload_json_to_minio(test_json, path_to_object, bucket_name, None)
+        mock_get_storage.assert_called_once_with(bucket_name)
+        mock_get_storage().upload_obj.assert_called_once_with(
+            target_path=path_to_object, file=ANY
+        )
+
+
+def test_upload_pages_to_minio(moto_s3: boto3.resource):
+    pages = [PageSchema(page_num=1, size={}, objs=[])]
+    pages_sha = {"1": "sha1"}
+    s3_path = "path"
+    with patch(
+        "annotation.annotations.main.upload_json_to_minio"
+    ) as mock_upload_json:
+        upload_pages_to_minio(pages, pages_sha, s3_path, TEST_TENANT, moto_s3)
+        path_to_object = f"{s3_path}/{pages_sha[str(pages[0].page_num)]}.json"
+        mock_upload_json.assert_called_once_with(
+            json.dumps(pages[0].dict()), path_to_object, TEST_TENANT, moto_s3
+        )
+
+def test_update_pages_array():
+    resulting_set = update_pages_array(
+        {1, 2, 3},
+        {4},
+        {1,2}
+    )
+    assert resulting_set == {3, 4}
+
+
+def test_kafka_message_needed(annotated_doc: AnnotatedDoc):
+    db = Mock()
```

Comment:
In this test, there is no assertion or error handling, which means that this is not a proper test. It needs to be corrected.